<launch>	
	
	<node pkg="tf" type="static_transform_publisher" name="base_to_dockcam_broadcaster" 
		args="-0.157 -0.003 0.080 3.14159 0 0 /base_link /dockcam 100" />
		
	<node pkg="autocrawler" type="docktargettostate.py" name="docktargettostate" required="true" output="screen"/>

	<arg name="dockdevice" default="/dev/video3" />
	<node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen" required="true" >
		<param name="video_device" value="$(arg dockdevice)" />
		<param name="image_width" value="640" />
		<param name="image_height" value="480" />
		<param name="pixel_format" value="yuyv" />
		<param name="camera_frame_id" value="usb_cam" />
		<param name="io_method" value="mmap"/>
		<param name="camera_info_url" value="file://$(find autocrawler)/dockcam_ost.yaml"/>
	</node>
	
	<arg name="peerid" default=""/>
	<arg name="webrtcserver" default="wss://127.0.0.1:8443"/>
	<arg name="turnserverlogin" default="user:pass"/>
	<arg name="turnserverport" default="3478"/>
	<node name="webrtcrs" pkg="autocrawler" type="webrtcrs" output="screen" 
		  args="--peer-id=$(arg peerid) --video-width=640 --server=$(arg webrtcserver)
			--video-height=480 --video-bitrate=256
			--turnserver-login=$(arg turnserverlogin) --turnserver-port=$(arg turnserverport)"  
		  respawn="true">
<!--
		<remap from="image_raw" to="/usb_cam/image_raw"/>
-->
		<remap from="image_raw" to="/fiducial_images"/>

	</node>

	<node pkg="autocrawler" type="webrtc_status_listener.py" name="webrtc_status_listener" required="true" output="screen" />
	
	<arg name="camera" default="/usb_cam"/>
	<arg name="image" default="image_raw"/>
	<arg name="transport" default="compressed"/>
	<arg name="fiducial_len" default="0.065"/>
	<arg name="dictionary" default="1"/>
	<arg name="do_pose_estimation" default="true"/>
	<arg name="ignore_fiducials" default="0-16,18-999" />
	<arg name="fiducial_len_override" default="" />
	<node pkg="aruco_detect" name="aruco_detect"
	type="aruco_detect" respawn="false">
		<param name="image_transport" value="$(arg transport)"/>
		<param name="publish_images" value="true" />
		<param name="fiducial_len" value="$(arg fiducial_len)"/>
		<param name="dictionary" value="$(arg dictionary)"/>
		<param name="do_pose_estimation" value="$(arg do_pose_estimation)"/>
		<param name="ignore_fiducials" value="$(arg ignore_fiducials)"/>
		<param name="fiducial_len_override" value="$(arg fiducial_len_override)"/>
		<remap from="/camera/compressed" 
			to="$(arg camera)/$(arg image)/$(arg transport)"/>
		<remap from="/camera_info" to="$(arg camera)/camera_info"/>
	</node>
	
</launch>
